code/tests/_game_test.dm

```diff
@@ -94,3 +94,67 @@
 		instance = new type()
 	LAZYADD(allocated, instance)
 	return instance
+
+/datum/game_test/proc/puppeteer(carbon_type = /mob/living/carbon/human, turf/initial_location)
+	TEST_ASSERT(ispath(carbon_type, /mob/living/carbon/human), "unexpected puppeteer carbon type [carbon_type]")
+	var/datum/test_puppeteer/puppeteer = new
+
+	if(!initial_location)
+		initial_location = locate(179, 136, 1) // Center of admin testing area
+	puppeteer.origin_test = src
+	puppeteer.puppet = allocate(carbon_type, initial_location)
+	var/datum/mind/new_mind = new("interaction_test_[puppeteer.puppet.UID()]")
+	new_mind.transfer_to(puppeteer.puppet)
+
+	return puppeteer
```

Comment:
can we get rid of this function and move all this behavior to /New?

Reply:
If we did we'd need to pass in the unit test anyway to perform the allocation, so hanging this proc off the test feels cleaner than creating a test puppeteer and passing in the test explicitly.

Reply:
I would prefer it that way, so that all the allocation of data is handled in /New, similar to how components initialize

Reply:
Done, PTAL